ltsa/lto_dump.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/ksh
+
+# Usage: KEYCLOAK_CLIENT_SECRET=your-secret ./lto_dump.sh [type]
+# type: 1 or 2 (required)
+
+# Configuration
+API_URL="${API_URL:-http://localhost:4007}"
+OUTPUT_FILE="ltodump.lis"
+
+# Check if type parameter is provided
+if [[ $# -eq 0 ]]; then
+    print "Error: Type parameter is required"
+    print "Usage: KEYCLOAK_CLIENT_SECRET=your-secret ./lto_dump.sh [type]"
+    print " type: 1 (PIDs < '025') or 2 (PIDs >= '025')"
+    exit 1
+fi
+
+TYPE=$1
+
+# Validate type parameter
+if [[ "$TYPE" != "1" ]] && [[ "$TYPE" != "2" ]]; then
+    print "Error: Type parameter must be 1 or 2"
+    print "  1: PIDs < '025'"
+    print "  2: PIDs >= '025'"
+    exit 1
+fi
+
+# Get the directory where this script is located
+SCRIPT_DIR=$(dirname $0)
+
+print "API URL: $API_URL"
+print "Type: $TYPE"
+print "Output file: $OUTPUT_FILE"
+
+# Step 1: Get access token
+print "\nGetting access token..."
+ACCESS_TOKEN=$("$SCRIPT_DIR/get_keycloak_token.sh")
+
+# Check if token retrieval was successful
+if [[ $? -ne 0 ]] || [[ -z "$ACCESS_TOKEN" ]]; then
+    print "Failed to get access token"
+    exit 1
+fi
+
+print "\nCalling API endpoint..."
+
+API_RESPONSE=$(curl -s -X GET \
```

Comment:
I would recommend a retry with backoff, for 3 or 5 times and finally fail, since it is a GET call and inherently idempotent